### PR TITLE
test: Re-group tests into Gradle suites

### PR DESCRIPTION
**Pull request by valfirst**
## Change list

Provide clear separation for different types of tests:
- unit tests
- integration Android tests
- integration iOS tests
 
## Types of changes

What types of changes are you proposing/introducing to Java client?

- [x] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by SrinivasanTarget — APPROVED** [review]

**Comment by valfirst** [comment]
@mykola-mokhnach could you kindly review the PR? I'd like to proceed with fixing/adding tests, this PR is a kind of prerequisite.

File: src/xcuiTest/java/io/appium/java_client/ios/TestResources.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import java.nio.file.Path;
+
+import static io.appium.java_client.TestUtils.resourcePathToLocalPath;
+
+public class TestResources {
+    private TestResources() {
+    }
+
+    public static Path testAppZip() {
+        return resourcePathToLocalPath("TestApp.app.zip");
```
Comment (mykola-mokhnach) [inline src/xcuiTest/java/io/appium/java_client/ios/TestResources.java:12]: should these be constants?

Comment (mykola-mokhnach) [reply src/xcuiTest/java/io/appium/java_client/ios/TestResources.java:12]: I don't see any reason to have it as a dynamic call

File: src/xcuiTest/java/io/appium/java_client/ios/AppIOSTest.java [hunk 2]
```diff
@@ -20,7 +20,7 @@ public static void beforeClass() throws Exception {
                 .setPlatformVersion(PLATFORM_VERSION)
                 .setDeviceName(DEVICE_NAME)
                 .setCommandTimeouts(Duration.ofSeconds(240))
-                .setApp(testAppZip().toAbsolutePath().toString())
+                .setApp(resourcePathToLocalPath("TestApp.app.zip").toAbsolutePath().toString())
```
Comment (mykola-mokhnach) [inline src/xcuiTest/java/io/appium/java_client/ios/AppIOSTest.java:23]: would it also make sense to have constants for other file names in resources?

File: build.gradle [hunk 3]
```diff
@@ -199,51 +190,104 @@ processResources {
...
+                }
+            }
+            targets.configureEach {
+                testTask.configure {
+                    finalizedBy jacocoTestReport
+                }
+            }
+        }
+
+        xcuiTest(JvmTestSuite) {
```
[the remark this answers is not in the document]
Comment (valfirst) [reply build.gradle]: renamed to `e2eAndroidTest` and `e2eIosTest`

**Pull request merged**